.github/workflows/differential-shellcheck.yml

```diff
@@ -0,0 +1,26 @@
+name: Differential ShellCheck
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    permissions:
+      security-events: write
```

Comment:
Ugh, the documentation is horrible.  Best I could find is https://docs.github.com/en/rest/overview/permissions-required-for-github-apps#permission-on-security-events

I'm wondering why security-events ***write*** is needed for this workflow.  Seems the default (read) should be good enough, no?

Reply:
Yeah, GitHub documentation is sometimes confusing and even incomplete. Documentation for token permissions is here:

https://docs.github.com/en/rest/code-scanning#upload-an-analysis-as-sarif-data

> Uploads SARIF data containing the results of a code scanning analysis to make the results available in a repository. You must use an access token with the security_events scope to use this endpoint for private repositories. You can also use tokens with the public_repo scope for public repositories only. GitHub Apps must have the security_events write permission to use this endpoint.

It is required to have `security-events: write` to successfully "write" the SARIF report to GitHub for further processing.

You can also check other GitHub actions that upload SARIF (e.g. https://github.com/github/codeql-action#usage). They also require `security-events: write`

I hope this answers your question.

Reply:
It does, but it seems likely others may also question this.  People see "security" and "write" and tend to freak-out first, but then Github doesn't make it easy for them to figure it out either.  Maybe putting that URL into a comment above the permission would help alleviate some fears?